packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -661,39 +765,177 @@ abstract class TableBase extends Resource implements ITable {
+      throw new Error("The Operation dimension is not supported. Use the 'operations' property.");
+    }
+
+    if (props.operations.length > 5 || props.operations.length === 0) {
+      // you must pass at least 1 operation since this is an operation dimensionality metric.
+      // you cannot pass more than 5 because this would make this metric unusable for alarms, who have a 10 individual metric limit for math expressions.
+      // https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#alarms-on-metric-math-expressions
+      throw new Error("Number of operations for the 'metricSuccessfulRequestLatency' metric should be between 1 and 5");
+    }
+
+    const valueMetricNameMapper = (op: Operation) => `${op}Value`.toLowerCase();
+    const countMetricNameMapper = (op: Operation) => `${op}Count`.toLowerCase();
+
+    const values = this.createMetricsForOperations('SuccessfulRequestLatency', props.operations, props, valueMetricNameMapper);
+    const counts = this.createMetricsForOperations('SuccessfulRequestLatency', props.operations, { ...props, statistic: 'n' }, countMetricNameMapper);
+
+    const denominator = props.operations.map(op => `${valueMetricNameMapper(op)}`).join(' + ');;
+    const numerator = props.operations.map(op => `${valueMetricNameMapper(op)} * ${countMetricNameMapper(op)}`).join(' + ');
+
+    return new cloudwatch.MathExpression({
+      expression: `(${numerator}) / (${denominator})`,
+      usingMetrics: { ...values, ...counts },
+      color: props?.color,
+      label: props?.label ?? 'Weighted average over all operations',
+      period: props?.period,
+    });
+
+  }
+
+  /**
+   * Create a map of metrics that can be used in a math expression.
+   *
+   * Using the return value of this function as the `usingMetrics` property in `cloudwatch.MathExpression` allows you to
+   * use the keys of this map as metric names inside you expression.
+   *
+   * @param metricName The metric name.
+   * @param operations The list of operations to create metrics for.
+   * @param props Properties for the individual metrics.
+   * @param metricNameMapper Mapper function to allow controlling the individual metric name per operation.
+   */
+  private createMetricsForOperations(metricName: string, operations: Operation[],
+    props?: cloudwatch.MetricOptions, metricNameMapper?: (op: Operation) => string): Record<string, cloudwatch.IMetric> {
+
+    const metrics: Record<string, cloudwatch.IMetric> = {};
+
+    const mapper = metricNameMapper ?? (op => op.toLowerCase());
+
+    if (props?.dimensions?.Operation) {
+      throw new Error('Invalid properties. Operation dimension is not supported when calculating operational metrics');
+    }
+
+    for (const operation of operations) {
+
+      const metric = this.metric(metricName, {
+        ...props,
+        dimensions: {
+          TableName: this.tableName,
+          Operation: operation,
+          ...props?.dimensions,
```

Comment:
Not sure there is a use case for this, but I think this is the *correct* behavior, allowing users to pass additional dimensions that we are not aware of.

Reply:
Yeah I suppose that's fine.